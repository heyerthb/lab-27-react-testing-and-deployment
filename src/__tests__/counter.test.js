import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Counter from '../components/counter/counter';

Enzyme.configure({adapter: new Adapter()});

describe ('Testing our Counter component', () => {
  it ('Up counter increases counter', () => {
    let app = mount (<Counter/>);
    let upClick = app.find('.up');
    expect(upClick).toBeDefined();
    expect(app.state('count')).toBe(0);
    upClick.simulate('click');
    expect(app.state('count')).toBe(1);
  })

  it ('Down counter decreases counter', () => {
    let app = mount (<Counter/>);
    let downClick = app.find('.down');
    expect(downClick).toBeDefined();
    expect(app.state('count')).toBe(0);
    downClick.simulate('click');
    expect(app.state('count')).toBe(-1);
  })



  it('render correctly', () => {
    const renderTree = renderer.create(<Counter/>).toJSON();
    expect (renderTree).toMatchSnapshot();
  })
})


