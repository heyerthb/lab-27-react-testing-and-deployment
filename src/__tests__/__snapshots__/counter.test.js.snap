// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing our Counter component render correctly 1`] = `
<section
  className="counter"
>
  <a
    className="down"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
